### PR TITLE
Script examples

### PR DESCRIPTION
**Pull request by goncy**
Adds some script component examples similar to the code showcased on the documentation

### CODE REVIEW

File: docs/basic-features/script.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ description: Next.js helps you optimize loading third-party scripts with the bui
 <details>
   <summary><b>Examples</b></summary>
   <ul>
+    <li><a href="https://github.com/vercel/next.js/tree/canary/examples/script-component">Script Component</a></li>
```
Comment (housseindjirdeh) [inline docs/basic-features/script.md:10]: I like your idea of putting together all the different strategies into a single example! I may do the same and consolidate all the other examples into a separate, single example focused on showing different third-party providers (example: `script-component-recipes`)

**Review by housseindjirdeh — DISMISSED** [review]
This is amazing, thank you!

File: docs/basic-features/script.md (hunk 1 above)
Comment (goncy) [reply docs/basic-features/script.md:10]: Thanks! If you need any help with that just let me know 🙌

File: examples/script-component/pages/lazy.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+export default function Lazyload() {
+  const [log, setLog] = useState([])
+
+  function addLog(text) {
+    setLog(() => log.concat({time: new Date(), text}))
+  }
+
+  useEffect(() => {
+    addLog(`Page loaded window.FB is undefined`)
+  }, [])
```
Comment (styfle) [inline examples/script-component/pages/lazy.js:13]: @goncy Looks like lint is failing

```
/home/runner/work/next.js/next.js/examples/script-component/pages/lazy.js
  13:6  warning  React Hook useEffect has a missing dependency: 'addLog'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

✖ 1 problem (0 errors, 1 warning)
```

Comment (goncy) [reply examples/script-component/pages/lazy.js:13]: Oops, my bad, fixing it!

Comment (styfle) [reply examples/script-component/pages/lazy.js]: Looks like lint its still failing:

```
/home/runner/work/next.js/next.js/examples/script-component/pages/lazy.js
  7:3  warning  The 'addLog' function makes the dependencies of useEffect Hook (at line 13) change on every render. To fix this, wrap the definition of 'addLog' in its own useCallback() Hook  react-hooks/exhaustive-deps
```

Comment (goncy) [reply examples/script-component/pages/lazy.js]: Now it should be fine

File: examples/script-component/pages/_app.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import '../styles/global.css'
+
+const MyApp = ({ Component, pageProps, router }) => {
+  const pathname = router.pathname;
+
+  return (
+    <>
+      <Component {...pageProps} />
+      {pathname !== "/" && (
```
Comment (styfle) [inline examples/script-component/pages/_app.js:11]: I like this pattern, great idea 👍

**Review by styfle — DISMISSED** [review]
Thanks!

**Review by styfle — APPROVED** [review]

**Pull request merged**